velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -276,14 +313,31 @@ core::PlanNodePtr PlanBuilder::TableScanBuilder::build(core::PlanNodeId id) {
   }
 
   if (!tableHandle_) {
-    tableHandle_ = std::make_shared<HiveTableHandle>(
-        connectorId_,
-        tableName_,
-        true,
-        (subfieldFiltersMap_.empty()) ? std::move(filters)
-                                      : std::move(subfieldFiltersMap_),
-        remainingFilterExpr,
-        dataColumns_);
+    tableHandle_ = [&]() -> std::shared_ptr<connector::ConnectorTableHandle> {
+#ifdef VELOX_ENABLE_CUDF
+      if (facebook::velox::cudf_velox::cudfIsRegistered() &&
+          facebook::velox::connector::getAllConnectors().count(
+              cudf_velox::exec::test::kParquetConnectorId) > 0 &&
+          facebook::velox::cudf_velox::cudfTableScanEnabled()) {
+        return std::make_shared<
+            cudf_velox::connector::parquet::ParquetTableHandle>(
+            cudf_velox::exec::test::kParquetConnectorId,
+            tableName_,
+            subfieldFilterExpr != nullptr,
+            subfieldFilterExpr,
+            remainingFilterExpr,
+            dataColumns_);
+      }
+#endif
+      return std::make_shared<HiveTableHandle>(
```

Comment:
We want to avoid `VELOX_ENABLE_CUDF` in the PlanBuilder.
1) Can we convert `HiveTableHandle` -> `udf_velox::connector::parquet::ParquetTableHandle` on the cudf side?
2) We could adopt a factory for the TableHandle. Some related work here https://github.com/facebookincubator/velox/pull/13798/files